src/shogun/machine/Machine.cpp

```diff
@@ -310,3 +306,23 @@ CLatentLabels* CMachine::apply_locked_latent(SGVector<index_t> indices)
 			"for %s\n", get_name());
 	return NULL;
 }
+
+uint64_t CMachine::supports() const 
+{
+	return m_supports;
+}
+
+bool CMachine::supports(uint64_t property_mask)
+{
+	return (m_supports & property_mask) == property_mask;
+}
+
+void CMachine::add_support(uint64_t property_mask)
+{
+	m_supports |= property_mask;
```

Comment:
@gf712 will have some comments

Reply:
maybe we can write some sort of base class to handle the bit logical operations in shogun to avoid having to rewrite these?

Reply:
and that can then be used inside AnyParameterProperties and here? Just thinking that at some point someone will make a mistake.. But these look fine

Reply:
Very good idea